### DISCUSSION TITLE
Prediksi untuk data kategorik

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#583] [Q&A] [answered]
Saya punya data jenis dan jumlah barang yang terjual setiap hari. Saya ambil data 5 barang dengan penjualan tertinggi setiap harinya, seperti ini : 

<img width="219" alt="image" src="https://user-images.githubusercontent.com/106010284/222700951-4ae879e6-61f2-4caf-a0c7-bb9653764956.png">

Saya ingin melakukan prediksi beberapa hari kedepan untuk jenis barang yang paling banyak terjual tiap harinya seperti ini : 

<img width="188" alt="image" src="https://user-images.githubusercontent.com/106010284/222701155-fce876b9-1a38-457b-95c3-389b0dbb054b.png">

Saya sudah mencoba menggunakan model Markov Chain, akan tetapi model ini sangat berat. Apakah ada model lain yang cocok untuk kebutuhan saya dengan data yg seperti itu ?

Mohon bantuannya kak, terima kasih🙏

**Comment by idoali** [comment, marked as answer]
Terima kasih pak atas pertanyaannya

Apakah penjualan tiap barang itu berkaitan satu dengan yang lainnya? (Penjualan barang 1 mempengaruhi penjualan barang 2 dll). Jika iya, bapak bisa menggunakan model deep learning dengan LSTM yang di bagian outputnya neuron biasa sejumlah jumlah jenis barang bapak (pada kasus bapak terdapat 6 jenis barang, maka outputnya pun juga 6 neuron). 

Jika ternyata penjualan jenis barang satu tidak mempengaruhi penjualan jenis barang yang lain, maka bapak bisa membuat model yang berbeda-beda untuk setiap jenis barang (penjualan jenis barang 1 akan diprediksi oleh model 1, penjualan jenis barang 2 akan diprediksi oleh model 2 dan seterusnya). 

Kemudian jika bapak hanya ingin menampilkan 5 jenis barang dengan penjualan terbanyak setiap harinya, bapak bisa melakukan filtering biasa dari hasil prediksi bapak. 

Semoga jawabannya bisa membantu pak

**Reply by yohanespm77** [reply]
Terima kasih atas arahannya, Mas Ido. Data penjualan jenis barangnya tidak saling mempengaruhi satu sama lain mas. 
Kalau 1 jenis barang diprediksi oleh 1 model, berarti ini bisa menggunakan model time series ya mas ? Apakah model time series bisa digunakan untuk data yg targetnya berupa kategorik ? Jika bukan, model apa ya kira-kira yang cocok digunakan mas ?

Terima kasih sebelumnya, mas.

**Reply by idoali** [reply]
Hasil prediksi dari model time series bukan kategori pak. Tapi pada kasus yang bapak hadapi sendiri, yang diprediksi adalah jumlah kan ya pak bukan kategori?

**Reply by yohanespm77** [reply]
Yang mau diprediksi itu kategori mas (jenis barang yang terjual) bukan jumlah barangnya. 
Kolom "jumlah" itu hanya tambahan aja, cuma mau mengurutkan jenis barang dari yang paling banyak terjual hingga yang paling sedikit.

**Reply by idoali** [reply]
Baik pak, berarti solusinya tetap sama seperti yang ada diatas

**Reply by yohanespm77** [reply]
Model yang cocok digunakan model apa ya mas ?
